functions/routes/ecom/webhook.js

```diff
@@ -231,6 +227,112 @@ exports.post = async ({ appSdk, admin }, req, res) => {
+            const isVariation = trigger.subresource === 'variations'
+            let query = 'status!=cancelled&transactions.type=recurrence'
+            query += '&transactions.app.intermediator.code=galaxpay_app'
+            if (sku) {
+              query += `&items.sku=${sku}`
+            } else {
+              query += `&items.product_id=${resourceId}`
+            }
+            try {
+              const { result } = await getOrderWithQueryString(appSdk, storeId, query, auth)
+
+              if (result && result.length) {
+                const galaxPaySubscriptions = await getSubscriptionsByListMyIds(
+                  galaxpayAxios,
+                  result.reduce((orderIds, order) => {
+                    orderIds.push(order._id)
+                    return orderIds
+                  }, [])
+                )
+
+                if (galaxPaySubscriptions && galaxPaySubscriptions.length) {
+                  galaxPaySubscriptions.forEach(async subscription => {
+                    try {
+                      const order = await findOrderById(appSdk, storeId, subscription.myId, auth)
+                      let product = { sku, price }
+
+                      if (!sku) {
+                        product = await getProductsById(appSdk, storeId, resourceId, auth)
+                      }
+
+                      if (order) {
+                        const { plan } = await getDocSubscription(order._id, collectionSubscription)
+                        order.items.forEach(item => {
+                          if (isVariation) {
+                            let variationProduct
+
+                            if (!sku) {
+                              variationProduct = product.variations
+                                .find(variationFind => variationFind.sku === item.sku)
+                            } else if (sku === item.sku) {
+                              variationProduct = product
+                            }
+
+                            if (variationProduct) {
+                              const newItem = {
+                                sku: variationProduct.sku,
+                                price: variationProduct.price || item.final_price || item.price
+                              }
+                              const isRemoveItem = quantity < item.quantity
+                              checkItemsAndRecalculeteOrder(order.amount, order.items, plan, newItem, isRemoveItem)
+                            }
+                          } else {
+                            if (item.sku === product.sku) {
+                              const newItem = {
+                                sku: product.sku,
+                                price: product.price
+                              }
+                              const isRemoveItem = quantity < item.quantity
+                              checkItemsAndRecalculeteOrder(order.amount, order.items, plan, newItem, isRemoveItem)
```

Comment:
Ideal na verdade era só removermos o item se o estoque for zerado realmente, se a quantidade original é 3 mas só tem 2 faz sentido tentar vender os 2 pelo menos...